### PR TITLE
Closes issue #28

### PR DESCRIPTION
**Pull request by zeuslawyer**
### What

Issue: https://github.com/orgs/smartcontractkit/projects/17/views/1?pane=issue&itemId=74252721

### Why

Planned upgrade as per the board.

NOTES: 
- have added /lib to gitignore and added forge install commands for the dependencies.  Tests are still passing.

### CODE REVIEW

**Comment by zeuslawyer** [comment]
https://github.com/smartcontractkit/foundry-starter-kit/issues/28

File: README.md [hunk 1]
```diff
@@ -37,45 +36,48 @@ Foundry Starter Kit is a repo that shows developers how to quickly build, test,
...
 
--   [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)  
-    -   You'll know you've done it right if you can run `git --version`
--   [Foundry / Foundryup](https://github.com/gakonst/foundry)
-    -   This will install `forge`, `cast`, and `anvil`
-    -   You can test you've installed them right by running `forge --version` and get an output like: `forge 0.2.0 (f016135 2022-07-04T00:15:02.930499Z)`
-    -   To get the latest of each, just run `foundryup`
-
-And you probably already have `make` installed... but if not [try looking here.](https://askubuntu.com/questions/161104/how-do-i-install-make)
+- [Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
```
Comment (zeuslawyer) [inline README.md:39]: Note several format changes without actual content changes.

File: README.md [hunk 2]
```diff
@@ -37,45 +36,48 @@ Foundry Starter Kit is a repo that shows developers how to quickly build, test,
...
-make test
 ```
 
-## Testing
+## Install dependencies as follows:
 
-```
-make test
-```
+You can add the `--no-commit` flag to each of these `install` commands if your workspace has uncommitted changes.
```
Comment (thodges-gh) [inline README.md]: If I follow these instructions from a clean repo, only having checked out this branch, I receive the following error message when running this command (and any of the `forge install` commands):
```
$ forge install foundry-rs/forge-std
Installing forge-std in /Users/thomas/workspace/foundry-starter-kit/lib/forge-std (url: Some("https://github.com/foundry-rs/forge-std"), tag: None)
fatal: 'lib/forge-std' already exists and is not a valid git repo
Error: 
git submodule exited with code 128
```

If I remove the `lib/forge-std` directory, I can then run the command with the `--no-commit` flag, but this seems like the wrong approach.

The better fix I found would be simply to change the instructions to say to run `forge install`, then all these components are installed without issue.

File: lib/solmate
Comment (thodges-gh) [inline lib/solmate:1]: Removing this file caused issues for me not being able to build contracts.

```
$ forge test
[⠃] Compiling...2024-08-19T10:54:49.951915Z ERROR foundry_compilers::artifacts: error="/Users/thomas/workspace/foundry-starter-kit-test1/lib/solmate/src/tokens/ERC20.sol": No such file or directory (os error 2)
[⠊] Compiling...
Error: 
failed to resolve file: "/Users/thomas/workspace/foundry-starter-kit-test1/lib/solmate/src/tokens/ERC20.sol": No such file or directory (os error 2); check configured remappings
	--> /Users/thomas/workspace/foundry-starter-kit-test1/test/mocks/LinkToken.sol
	@solmate/tokens/ERC20.sol
```

If we add this back (with the new reference to `transmissions11`), the `forge install` command should be complete.

**Review by thodges-gh — CHANGES_REQUESTED** [review]

File: lib/solmate
Comment (zeuslawyer) [reply lib/solmate:1]: have reinstated the lib director and rolled back gitignore changes in commit sha 2af540ee4cc2746a3c72c76e2b17a9a0c39feb47.  The dep install and forge test steps now work on a fresh clone.

File: README.md (hunk 2 above)
Comment (zeuslawyer) [reply README.md]: fixed

**Review by thodges-gh — APPROVED** [review]

**Pull request merged**